src/platform/silabs/efr32/CHIPCryptoPALPsaEfr32.cpp

```diff
@@ -161,13 +161,14 @@ CHIP_ERROR AES_CCM_encrypt(const uint8_t * plaintext, size_t plaintext_length, c
         // Add the aligned part of the plaintext
         status = psa_aead_update(&operation, plaintext, block_aligned_length, ciphertext, block_aligned_length, &out_length);
         VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
-        VerifyOrReturnError(out_length == block_aligned_length, CHIP_ERROR_INTERNAL);
+        VerifyOrReturnError(out_length <= block_aligned_length, CHIP_ERROR_INTERNAL);
```

Comment:
Might be worth documenting why this might be the case even with a symmetric cypher.

Reply:
There is no requirement that `psa_aead_update` should return `PSA_AEAD_UPDATE_OUTPUT_SIZE`. I believe that's just a wrong assumption.

Reply:
My point is that with a symmetric cypher (where the sum of all the psa_aead_update()/finish() calls has to end up returning the same amount of data as you put in), it's not obvious that any given call does _not_ do that.

Reply:
I disagree. It is wrong to assume that the output length should be equal to the input length. These functions has an explicit `output_length` parameter, which returns the number of bytes actually written. Such parameter would be unnecessary if the output length was always equal to the input length upon success.

Reply:
The code around all this stuff absolutely assumes that _by the time you are done decrypting the whole message_ the output and input lengths are the same, because it does in-place decryption, in general.  And for a symmetric cypher that's a correct assumption.  But what's not OK to assume is that this is the case for any single call to update.  But OK, either way.